ffi/capi/src/timezone.rs

```diff
@@ -34,6 +35,28 @@ pub mod ffi {
             Box::new(Self(icu_timezone::TimeZoneInfo::utc().into()))
         }
 
+        /// Creates a time zone.
+        #[diplomat::attr(auto, constructor)]
+        pub fn from_parts(
+            bcp47_id: &DiplomatStr,
+            offset_seconds: i32,
+            dst: bool,
+        ) -> Box<TimeZoneInfo> {
+            Box::new(Self(TimeZoneInfoBuilder {
+                time_zone_id: TinyAsciiStr::try_from_utf8(bcp47_id)
+                    .ok()
+                    .map(TimeZoneBcp47Id)
+                    .unwrap_or(TimeZoneBcp47Id::unknown()),
+                offset: UtcOffset::try_from_seconds(offset_seconds).ok(),
+                zone_variant: Some(if dst {
+                    ZoneVariant::daylight()
+                } else {
+                    ZoneVariant::standard()
+                }),
+                local_time: None,
```

Comment:
Issue: doesn't this hit the `.ok_or(Error::DateTimeZoneInfoMissingFieldsError)?` down below?

Reply:
yeah fixed in the next PR